sdk/keyvault/azure-security-keyvault-keys/src/main/java/com/azure/security/keyvault/keys/cryptography/implementation/AesKeyCryptographyClient.java

```diff
@@ -380,21 +341,19 @@ public Mono<WrapResult> wrapKeyAsync(KeyWrapAlgorithm algorithm,
         LocalKeyWrapAlgorithm localKeyWrapAlgorithm = (LocalKeyWrapAlgorithm) baseAlgorithm;
 
         return Mono.fromCallable(() -> {
-            byte[] encrypted = localKeyWrapAlgorithm.createEncryptor(key, null, null)
+            byte[] encrypted = localKeyWrapAlgorithm.createEncryptor(aesKey, null, null)
                 .doFinal(keyToWrap);
 
             return new WrapResult(encrypted, algorithm, jsonWebKey.getId());
         });
     }
 
     @Override
-    public WrapResult wrapKey(KeyWrapAlgorithm algorithm, byte[] keyToWrap, JsonWebKey jsonWebKey, Context context) {
+    public WrapResult wrapKey(KeyWrapAlgorithm algorithm, byte[] keyToWrap, Context context) {
         Objects.requireNonNull(algorithm, "Key wrap algorithm cannot be null.");
         Objects.requireNonNull(keyToWrap, "Key content to be wrapped cannot be null.");
 
-        this.key = getKey(jsonWebKey);
-
-        if (this.key == null || this.key.length == 0) {
+        if (this.aesKey == null || this.aesKey.length == 0) {
```

Comment:
Yes, we should move this to the constructor. I think this was in place in these methods because we used to take the key in the method parameters, but that key is really the one we set in the constructor as well.